test/UnitTest/Microsoft.AspNetCore.OData.Test/Extensions/Endpoint/ODataEndpointRouteValueTransformerTests.cs

```diff
@@ -104,6 +107,96 @@ public void TransformAsyncReturnsCorrectRouteValues()
+                c => c.AddService(ServiceLifetime.Singleton, b => GetEdmModel()));
+
+            HttpContext httpContext = new DefaultHttpContext
+            {
+                RequestServices = builder.ServiceProvider
+            };
+
+            var httpMethod = isOptionsRequest ? HttpMethod.Options : HttpMethod.Get;
+
+            HttpRequest request = SetHttpRequest(httpContext, httpMethod, "http://localhost:123/Customers(1)");
+            RouteValueDictionary values = new RouteValueDictionary();
+            values.Add("ODataEndpointPath_odata", "Customers(1)");
+
+            ActionDescriptor actionDescriptor = new ControllerActionDescriptor
+            {
+                ControllerName = "Customers",
+                ActionName = "Get",
+                EndpointMetadata = GenerateMetadata(includeMetadata)
+            };
+
+            IActionSelector actionSelector = new MockActionSelector(actionDescriptor);
+
+            // Act
+            ODataEndpointRouteValueTransformer transformer = new ODataEndpointRouteValueTransformer(actionSelector);
+            ValueTask<RouteValueDictionary> actual = transformer.TransformAsync(httpContext, values);
+            var endpoint = httpContext.GetEndpoint();
+            // Assert the metadata is empty when options requests are sent and the controller is not annotated
+            Assert.Null(endpoint);
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        public void TransformAsyncReturnsCorrectRouteValuesForOptionRequests(bool isOptionsRequest, bool includeMetadata)
+        {
+            // Arrange
+            IEndpointRouteBuilder builder = EndpointRouteBuilderFactory.Create("odata",
+                c => c.AddService(ServiceLifetime.Singleton, b => GetEdmModel()));
+
+            HttpContext httpContext = new DefaultHttpContext
+            {
+                RequestServices = builder.ServiceProvider
+            };
+
+            var httpMethod = isOptionsRequest ? HttpMethod.Options : HttpMethod.Get;
+
+            HttpRequest request = SetHttpRequest(httpContext, httpMethod, "http://localhost:123/Customers(1)");
+            RouteValueDictionary values = new RouteValueDictionary();
+            values.Add("ODataEndpointPath_odata", "Customers(1)");
+
+            ActionDescriptor actionDescriptor = new ControllerActionDescriptor
+            {
+                ControllerName = "Customers",
+                ActionName = "Get",
+                EndpointMetadata = GenerateMetadata(includeMetadata)
+            };
+
+            IActionSelector actionSelector = new MockActionSelector(actionDescriptor);
+
+            // Act
+            ODataEndpointRouteValueTransformer transformer = new ODataEndpointRouteValueTransformer(actionSelector);
+            ValueTask<RouteValueDictionary> actual = transformer.TransformAsync(httpContext, values);
+            var endpoint = httpContext.GetEndpoint();
```

Comment:
Can all this repeated logic be encapsulated into some build/helper method that the test calls - keep things DRY

Reply:
Updated